### PR TITLE
Fix Flying Tera Type Pokemon using Roost

### PR DESCRIPTION
**Pull request by MathyFurret**
https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9403448

### CODE REVIEW

File: test/sim/moves/roost.js [hunk 1]
```diff
@@ -78,6 +78,16 @@ describe('Roost', function () {
...
 		assert.equal(battle.p1.active[0].hp, battle.p1.active[0].maxhp); // Immune to Astonish
 	});
+
+	it('should not remove Flying type during Terastallization', function () {
+		battle = common.createBattle([[
+			{species: "Dudunsparce", ability: "runaway", moves: ['roost'], teraType: "Flying"},
+		], [
+			{species: "Bonsly", ability: "sturdy", moves: ['earthquake']},
+		]]);
+		battle.makeChoices('move roost terastallize', 'auto');
```
Comment (DaWoblefet) [inline test/sim/moves/roost.js:88]: Roost won't remove the user's Flying-type at 100% HP either, so this test needs modified.

File: sim/pokemon.ts [hunk 2]
```diff
@@ -1968,7 +1968,7 @@ export class Pokemon {
 		const item = (this.ignoringItem() ? '' : this.item);
 		if (item === 'ironball') return true;
 		// If a Fire/Flying type uses Burn Up and Roost, it becomes ???/Flying-type, but it's still grounded.
```
Comment (MathyFurret) [inline sim/pokemon.ts:1970]: I couldn't find a unit test for this interaction but I don't trust myself to write one, it's beyond my comprehension.

Comment (MathyFurret) [reply sim/pokemon.ts:1970]: Actually, has Ground immunity for Tera Flying Roost actually been tested?

Comment (MathyFurret) [reply sim/pokemon.ts:1970]: https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9410489 It has now

**Pull request merged**